test/node/features/express.feature

```diff
@@ -28,17 +28,20 @@ Scenario: a synchronous thrown error in a route
   And the event "metaData.request.connection" is not null
 
 Scenario: an asynchronous thrown error in a route
-  Then I open the URL "http://express/async"
+  Then I open the URL "http://express/async" tolerating any error
   And I wait to receive an error
   Then the error is valid for the error reporting API version "4" for the "Bugsnag Node" notifier
   And the event "unhandled" is true
   And the event "severity" equals "error"
-  And the event "severityReason.type" equals "unhandledErrorMiddleware"
+  And the event "severityReason.type" equals "unhandledException"
```

Comment:
These `unhandledErrorMiddleware` &rarr; `unhandledException` look like a regression to me, since it affects what's shown in the dashboard:

<img width="339" alt="Screenshot 2023-02-21 at 10 02 20" src="https://user-images.githubusercontent.com/282732/220313197-398634d7-39b5-4a65-a4ea-d248b596640d.png">

I _think_ adding an `onError` in the server plugins that sets the handled state for unhandled events would fix this and be safe?